pkg/segment/results/mresults/metricresults.go

```diff
@@ -399,45 +407,32 @@ func (res *MetricsResult) GetMetricTagsResultSet(mQuery *structs.MetricsQuery) (
 	uniqueTagKeyValues := make(map[string]bool)
 	tagKeyValueSet := make([]string, 0)
 
-	for _, series := range res.AllSeries {
-		seriesStr := removeTrailingComma(series.grpID.String())
-		tagKeyValues := strings.Split(seriesStr, tsidtracker.TAG_VALUE_DELIMITER_STR)
-
-		for _, tkVal := range tagKeyValues {
-			if _, ok := uniqueTagKeyValues[tkVal]; !ok {
-				uniqueTagKeyValues[tkVal] = true
-				tagKeyValueSet = append(tagKeyValueSet, tkVal)
+	for _, tsidInfo := range res.AllSeriesTagsOnlyMap {
+		for tagKey, tagValue := range tsidInfo.TagKeyTagValue {
+			tagKeyValue := fmt.Sprintf("%s: %s", tagKey, tagValue)
```

Comment:
I don;t think we put a space between `"%s: %s"` in here. Please confirm this with playgorund

Reply:
I confirmed this with playground and there shouldn't be a space. I just pushed a commit to remove the space